### ISSUE TITLE
object detection accuracy bug: mAP does not match with the tensorflow released data

### ISSUE THREAD

**Issue by xhzhao** [tensorflow/tensorrt#40]
we tried to repeate the tf pretrained model accuracy on tensorrt, and we found that the "ssd_inception_v2_coco" accuracy on tensorrt is not the same as the tensorflow released data:(24 vs 27)
could you help to find out why?

see tensorflow release data [here](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md)

![image](https://user-images.githubusercontent.com/17486215/54337078-bd6b9c00-4668-11e9-9a09-7961f540033b.png)

**Comment by pooyadavoodi** [comment]
The difference could be also because that table from tensorflow/models is kind of old. Perhaps if you try to reproduce those numbers, you get different results.

Also make sure you use the same validation data set.

**Comment by xhzhao** [comment]
we aligned the validation dataset with tensorflow model zoo, but the accuracy still does not match.
https://github.com/tensorflow/models/blob/master/research/object_detection/data/mscoco_minival_ids.txt

my config:
{
  "source_model": {
    "model_name": "ssd_inception_v2_coco",
    "output_dir": "models"
  },
  "optimization_config": {
    "use_trt": false,
    "precision_mode": "FP32",
    "force_nms_cpu": true,
    "replace_relu6": true,
    "remove_assert": true,
    "override_nms_score_threshold": 0.3,
    "max_batch_size": 1
  },
  "benchmark_config": {
    "images_dir": "/home/xhzhao/code/ngraph/dataset/coco_val/val2014/",
    "annotation_path": "/home/xhzhao/code/ngraph/dataset/coco_val/instances_val2014.json",
    "batch_size": 1,
    "image_shape": [600, 600],
    "output_path": "stats/ssd_inception_v2_coco_trt_fp16.json"
  },
  "assertions": [
    "statistics['map'] > (0.268 - 0.005)"
  ]
}

log output:
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.198
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.314
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.208
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.022
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.150
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.409
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.188
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.240
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.241
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.030
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.180
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.496
{
    "avg_latency_ms": 24.382473739357057,
    "avg_throughput_fps": 41.013065806602164,
    "map": 0.1982614882543663
}

**Comment by pooyadavoodi** [comment]
Could you remove the following items from the config file and retry:

```
"force_nms_cpu": true,
"replace_relu6": true,
"remove_assert": true,
"override_nms_score_threshold": 0.3,
```

Also, have you tried to use the scripts from tensorflow/models to reproduce that mAP. Perhaps the model has changed and that mAP can't be reproduced by this model?

**Comment by lvniqi** [comment]
@xhzhao xhzhao 
i think you shouldn't set image_shape to [600,600]. 
if you reshape the image, the Aspect Ratio may be wrong.

**Comment by pooyadavoodi** [comment]
I set image_shape to [640x640] and I get mAP: 0.276.

**Comment by lvniqi** [comment]
> I set image_shape to [640x640] and I get mAP: 0.276 for ssd_inception_v2.

ok.  you are right. it doesn't a matter about image_shape.

**Comment by pooyadavoodi** [comment]
Closing. Please reopen in case the issue remains.
